source/Missions/Services/Agents/Packets/MovementHandler.cs

```diff
@@ -131,13 +193,18 @@ public void Handle_PeerDisconnect(MessagePayload<PeerDisconnected> payload)
 
             if (_agentRegistry.OtherAgents.TryGetValue(peer, out AgentGroupController controller))
             {
-                foreach (var agent in controller.ControlledAgents.Values)
+                foreach (var kv in controller.ControlledAgents)
                 {
+                    var agent = kv.Value;
+                    var guid = kv.Key;
```

Comment:
```suggestion
                foreach (var guid in controller.ControlledAgents.Keys)
                {
                    var guid = kv.Key;
```

Reply:
could you please explain why you suggest this? I don't seem to understand. :)